src/views/endpoint/autopilot/AutopilotListProfiles.js

```diff
@@ -1,27 +1,58 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { useSelector } from 'react-redux'
 import TenantSelector from '../../../components/cipp/TenantSelector'
 import CippDatatable from '../../../components/cipp/CippDatatable'
 import { CCard, CCardHeader, CCardTitle, CCardBody } from '@coreui/react'
 import { cellBooleanFormatter } from '../../../components/cipp'
-//future version dropdown:
-// const dropdown = (row, rowIndex, formatExtraData) => {
-//   return (
-//     <CDropdown>
-//       <CDropdownToggle size="sm" color="link">
-//         <FontAwesomeIcon icon={faBars} />
-//       </CDropdownToggle>
-//       <CDropdownMenu style={{ position: 'fixed', right: 0, zIndex: 1000 }}>
-//         <CDropdownItem href="#">
-//           <Link className="dropdown-item" to={`/endpoint/autopilot/AutopilotEditProfile}`}>
-//             <FontAwesomeIcon icon={faUser} className="me-2" />
-//             Edit Profile
-//           </Link>
-//         </CDropdownItem>
-//       </CDropdownMenu>
-//     </CDropdown>
-//   )
-// }
+import { CButton } from '@coreui/react'
+import { faCopy, faEye } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { CippOffcanvas } from '../../../components/cipp'
+
+const Offcanvas = (row, rowIndex, formatExtraData) => {
+  const [visible, setVisible] = useState(false)
+
+  const jsonContent = <pre>{JSON.stringify(row, null, 2)}</pre>
```

Comment:
Why wrap it in `<pre>` here? Just have the JSON here and wrap it during the render / return call :-)

Reply:
Having it in `<pre>` here is making the next section (getting the content for the copy) more complex than it needs to be :-)

Reply:
So I did do that originally, but got an error;

![image](https://user-images.githubusercontent.com/30923966/149029993-d23773b3-50b9-480b-b61c-20f2fb1444a3.png)

Reply:
Use:

```js
const jsonContent = (JSON.stringify(data))
```

Reply:
So that does work, but causes formatting issues which I think looks crappier than it did:

![image](https://user-images.githubusercontent.com/30923966/149031450-85171664-7033-4287-96b6-98c94875a2b9.png)

Reply:
You still need to wrap it in a `<pre` just inside `<CippOffCanvas`

Reply:
Yep, I did

`      <CippOffcanvas
        visible={visible}
        id={data.id}
        placement="end"
        className="cipp-offcanvas"
        hideFunction={() => setVisible(false)}
        title="JSON (Raw)"
      >
        <CButton size="sm" variant="ghost" color="info" onClick={CopyToClipboard}>
          <FontAwesomeIcon icon={faCopy} />
        </CButton>
        <pre>{jsonContent}</pre>
      </CippOffcanvas>
`

Reply:
Sorry I overthought this...

`const jsonContent = JSON.stringify(error)` should work fine.